js/packages/teams-ai/src/PlannerImpl.ts

```diff
@@ -0,0 +1,80 @@
+/**
+ * @module teams-ai
+ */
+/**
+ * Copyright (c) Microsoft Corporation. All rights reserved.
+ * Licensed under the MIT License.
+ */
+
+import { TurnContext } from 'botbuilder';
+import { DefaultTurnState } from './DefaultTurnStateManager';
+import { Plan, Planner } from './Planner';
+import { TurnState } from './TurnState';
+import { PromptTemplate } from './Prompts';
+import { ConfiguredAIOptions } from './AI';
+import { AIApiFactory } from './AIApiFactory';
+
+/**
+ * A planner that uses OpenAI's textCompletion and chatCompletion API's to generate plans.
+ * @summary
+ * This planner can be configured to use different models for different prompts. The prompts model
+ * will determine which API is used to generate the plan. Any model that starts with 'gpt-' will
+ * use the chatCompletion API, otherwise the textCompletion API will be used.
+ * @template TState Optional. Type of the applications turn state.
+ * @template TOptions Optional. Type of the planner options.
+ */
+class PlannerImpl<TState extends TurnState = DefaultTurnState> implements Planner<TState> {
```

Comment:
Whats the point of a distinction between `ApiFactory` and `PlannerImpl` if the planner is just a call through for the factory methods?

Reply:
`ApiFactory` is the main extension point for different LLMs.
based on a prompt, you might want to call a different service, or run locally. `PlannerImpl` is intentionally small so it will simply call the correct factory.